lib/indexer/popularity_lookup.rb

```diff
@@ -16,37 +16,43 @@ def lookup_popularities(links)
             path_components: links,
           },
         },
-        _source: { includes: %w[rank_14] },
+        _source: { includes: %w[rank_14 vc_14] },
         sort: [
           { rank_14: { order: "asc" } },
         ],
         size: 10 * links.size,
       })
 
-      ranks = Hash.new(traffic_index_size)
+      default_rank = Hash.new(
+        rank: traffic_index_size,
+        view_count: 1,
+      )
 
-      results["hits"]["hits"].each do |hit|
+      ranks = results["hits"]["hits"].each_with_object(default_rank) do |hit, hsh|
         link = hit["_id"]
-        rank = Array(hit["_source"]["rank_14"]).first
+        rank = Array(hit.dig("_source", "rank_14")).first
+        view_count = hit.dig("_source", "vc_14") || 1
         next if rank.nil?
 
-        ranks[link] = [rank, ranks[link]].min
+        hsh[link] = { rank: rank, view_count: view_count }
       end
 
       Hash[links.map { |link|
-        if ranks[link].zero?
+        if ranks[link][:rank].zero?
           popularity_score = 0
         else
-          popularity_score = 1.0 / (ranks[link] + SearchConfig.popularity_rank_offset)
+          popularity_score = 1.0 / (ranks[link][:rank] + SearchConfig.popularity_rank_offset)
         end
 
-        popularity_rank = traffic_index_size - (ranks[link] || traffic_index_size)
+        popularity_rank = traffic_index_size - (ranks[link][:rank] || traffic_index_size)
```

Comment:
I'm not sure this is right if `ranks[link][:rank].zero?` holds:

```
traffic_index_size - (ranks[link][:rank] || traffic_index_size)
= traffic_index_size - (0 || traffic_index_size)         ; ranks[link][:rank].zero? == true
= traffic_index_size - 0                                 ; 0 || x == 0
= traffix_index_size                                     ; x - 0 = 0
```

So if something has a rank of zero (which, to be fair, I don't think should happen because [ranking starts from 1](https://github.com/alphagov/search-analytics/blob/master/analytics_fetcher/fetch.py#L64)) then it gets given the greatest possible `popularity_rank`.